lib/assert.js

```diff
@@ -278,7 +278,13 @@ function expectedException(actual, expected) {
     // Ignore.  The instanceof check doesn't work for arrow functions.
   }
 
-  return expected.call({}, actual) === true;
+  try {
+    return expected.call({}, actual) === true;
+  } catch (e) {
+    // Ignore. If expected is a class constructor, an error will be thrown
+  }
```

Comment:
I agree, this seems dangerous.  You could rethrow unless `e instanceof TypeError && /Class constructors cannot be invoked without 'new'/.test(e.message)` but that changes the throw site and can still swallow genuine TypeErrors from the `expected` callback.